pkg/build/oci/oci.go

```diff
@@ -56,10 +56,10 @@ const (
 )
 
 func BuildImageFromLayer(layer v1.Layer, ic types.ImageConfiguration, logger log.Logger, opts options.Options) (oci.SignedImage, error) {
```

Comment:
oci.go is hundreds of lines of critically important code, with no tests, and this is a pretty thorough refactor of this code.

I don't want to bog this change down by making you add tests for all of this code, but if there's anything you're adding in this PR that you think warrants a test, I'd really love to start adding them. 🙏

Reply:
@imjasonh I appreciate not wanting to bog this down, yet I agree that we really need them, as oct.go _is_ hundreds of lines of critically important code.

In that vein, I started by restructuring the oci _package_ into 6 separate files, each with many fewer functions. That should make it much easier to write tests. I did it as a separate commit, to make it easier to view separately. It doesn't change any functionality, just moves the single overly large file into dedicated ones for building images, building indexes, sbom, publishing types and constants.

Reply:
And I cleaned up some duplication, and I added inline documentation to pretty much every func in `package oci`, and created the 4 `*_test.go` files, and added the empty skeletons, so every primary func is defined.

We need a lot of those tests, but I am not convinced we can get that done in here. This was a big step forward., though